## 71% of end-end tests have passed
```
Total executed tests: 14
Failed tests: 1
Expected to fail tests: 3
Passed tests: 10
```
```
IDs of failed tests: 727230 
```
```
IDs of expected to fail tests: 741840,741839,741841 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/unblock-validation&PR_ID=22323&APK_NAME=22323.apk&TR_CASE_IDS=727230">Rerun failed tests</a></li><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814513">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: LogInButton </div><div>## Signed in successfully!</div></blockquote></p>

```
critical/wallet/test_wallet_testnet.py:191: in test_wallet_send_asset_from_drawer
    self._check_balances_after_tx(amount_to_send, None, None, eth_amount_sender, eth_amount_receiver)
critical/wallet/test_wallet_testnet.py:88: in _check_balances_after_tx
    self.loop.run_until_complete(
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:38: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/base_view.py:354: in reopen_app
    sign_in_view.sign_in(user_name, password)
../views/sign_in_view.py:134: in sign_in
    self.get_user_profile_by_name(user_name).click()
../views/sign_in_view.py:144: in get_user_profile_by_name
    if expected_element.is_element_displayed(10):
../views/base_element.py:206: in is_element_displayed
    return self.wait_for_visibility_of_element(sec, ignored_exceptions=ignored_exceptions)
../views/base_element.py:136: in wait_for_visibility_of_element
    .until(expected_conditions.visibility_of_element_located((self.by, self.locator)))
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:86: in until
    value = method(self._driver)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:152: in _predicate
    return _element_if_visible(driver.find_element(*locator))
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:175: in _element_if_visible
    return element if element.is_displayed() == visibility else False
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webelement.py:81: in is_displayed
    return self._execute(Command.IS_ELEMENT_DISPLAYED)['value']
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:395: in _execute
    return self._parent.execute(command, params)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: 'GET /element/00000000-0000-0014-ffff-ffff0000003c/attribute/displayed' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: 'GET /element/00000000-0000-0014-ffff-ffff0000003c/attribute/displayed' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E       at UIA2Proxy.command (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at runMicrotasks (<anonymous>)
E       at processTicksAndRejections (node:internal/process/task_queues:96:5)
E       at AndroidUiautomator2Driver.getAttribute (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/lib/commands/element.js:33:7)
E       at AndroidUiautomator2Driver.elementDisplayed (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/lib/commands/element.js:41:19)
```

<br/><br/></td></tr></tbody></table></details><h3>Expected to fail tests (3)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletCollectibles:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814520">test_wallet_send_collectible</a>, id: 741840 </b></td></tr><tr><td><p><blockquote><div>Device 1: Swiping right on element SlideButton </div><div>Device 1: Find `SlideButton` by `xpath`: `//*[@resource-id='slide-button-track']` </div></blockquote></p>

```
critical/wallet/test_collectibles.py:141: in test_wallet_send_collectible
    self.wallet_view.slide_button_track.slide()
../views/base_view.py:107: in slide
    self.swipe_right_on_element(width_percentage=1.3, start_x=100)
../views/base_element.py:288: in swipe_right_on_element
    location, size = self.get_element_coordinates()
../views/base_element.py:274: in get_element_coordinates
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: SlideButton by xpath: `//*[@resource-id='slide-button-track']` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814519">test_wallet_collectibles_balance</a>, id: 741839 </b></td></tr><tr><td><p><blockquote><div>Device 1: `Text` is `Double Spike` </div><div>Device 1: Click system back button </div></blockquote></p>

```
critical/wallet/test_collectibles.py:95: in test_wallet_collectibles_balance
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: BVL image doesn't match expected template
E    Device 1: BVL quantity 2 doesn't match expected 1
E    Device 1: BVL expanded image doesn't match expected template
E    Device 1: Glitch Punks image doesn't match expected template 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814521">test_wallet_collectible_send_from_expanded_info_view</a>, id: 741841 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `collectibles-tab` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/wallet/test_collectibles.py:153: in test_wallet_collectible_send_from_expanded_info_view
    self.wallet_view.get_collectible_element('Glitch Punks').wait_for_element().click()
../views/base_element.py:120: in wait_for_element
    raise TimeoutException(
 Device `1`: `CollectibleItemElement` by` xpath`: `//*[@content-desc='collectible-list-item']//*[contains(@text,'Glitch Punks')]/../..` is not found on the screen after wait_for_element 
```

<b>[[Collectibles list is not loaded sometimes]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (10)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814514">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814517">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814515">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814518">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814516">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814511">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESmj5tl2SZ5Rvucil4">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814508">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESmj5tl2SZ5Rvucil4">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814510">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESQLRNBn3OPrE5aQ9S">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESygLY5WRifr75DcPD">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814509">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESgibeQ5b1W4nG3RwF">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESRejrLnTvPSH2XEru">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3814512">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table></details>

--- a/src/status_im/contexts/keycard/unblock/events.cljs
+++ b/src/status_im/contexts/keycard/unblock/events.cljs
@@ -1,5 +1,6 @@
 (ns status-im.contexts.keycard.unblock.events
-  (:require [utils.re-frame :as rf]
+  (:require [utils.i18n :as i18n]
+            [utils.re-frame :as rf]
             [utils.security.core :as security]))
 
 (defn show-different-card
@@ -24,12 +25,15 @@
          [:dispatch [:open-modal :screen/keycard.unblock.ready-to-unblock]]]}))
 
 (rf/reg-event-fx :keycard/unblock.phrase-entered
- (fn [{:keys [db]} [{:keys [seed-phrase]}]]
-   {:db (assoc-in db [:keycard :unblock :masked-phrase] seed-phrase)
-    :fx [[:dispatch [:navigate-back]]
-         [:dispatch
-          [:open-modal :screen/keycard.pin.create
-           {:on-complete #(rf/dispatch [:keycard/unblock.pin-created (security/mask-data %)])}]]]}))
+ (fn [{:keys [db]} [{:keys [seed-phrase key-uid]}]]
+   (if (not= key-uid (get-in db [:keycard :application-info :key-uid]))
+     {:fx [[:dispatch
+            [:enter-seed-phrase/set-error (i18n/label :t/seed-phrase-dont-match-migrate-flow)]]]}
+     {:db (assoc-in db [:keycard :unblock :masked-phrase] seed-phrase)
+      :fx [[:dispatch [:navigate-back]]
+           [:dispatch
+            [:open-modal :screen/keycard.pin.create
+             {:on-complete #(rf/dispatch [:keycard/unblock.pin-created (security/mask-data %)])}]]]})))
 
 (rf/reg-event-fx :keycard/unblock.generate-and-load-key
  (fn [{:keys [db]}]

